### PR TITLE
Code cleanup to increase cpplint-cleanliness

### PR DESCRIPTION
**Pull request by tautschnig**
This patch series should go together with #333 as that one also helps cleanliness.

### CODE REVIEW

File: src/solvers/flattening/boolbv.cpp [hunk 1]
```diff
@@ -143,7 +143,7 @@ const bvt& boolbvt::convert_bv(const exprt &expr)
     bv_cache.insert(std::make_pair(expr, bvt()));
   if(!cache_result.second)
   {
-    //std::cerr << "Cache hit on " << expr << "\n";
+    // std::cerr << "Cache hit on " << expr << "\n";
```
Comment (peterschrammel) [inline src/solvers/flattening/boolbv.cpp]: Shouldn't this be guarded by #ifdef DEBUG? (several instances ahead)

File: src/util/ieee_float.cpp [hunk 2]
```diff
@@ -1804,18 +1804,18 @@ void ieee_floatt::next_representable(bool greater)
 
   unpack(old);
 
-  //sign change impossible (zero case caught earler)
+  // sign change impossible (zero case caught earler)
   set_sign(old_sign);
 
-  //mp_integer new_exp = exponent;
-  //mp_integer new_frac = fraction + dir;
+  // mp_integer new_exp = exponent;
```
Comment (peterschrammel) [inline src/util/ieee_float.cpp]: code commented out? -> use #if 0 or remove

**Comment by tautschnig** [comment]
I have updated the changes to address the comments by @peterschrammel.

**Comment by tautschnig** [comment]
It seems that the status of #451 should be clarified before merging (or attempting to rebase) this one.

**Comment by tautschnig** [comment]
At this point I would declare the code base lint clean for practical purposes, with the exception of src/cegis, which I haven't heavily touched. See
[all-issues.txt](https://github.com/diffblue/cbmc/files/737407/all-issues.txt) for any remaining ones. This file was generated using ` scripts/cpplint.py `find src -name "*.h" -o -name "*.cpp"` 2> all-issues`.

I will now work toward cleaning this commit series and do hope to find people to review it.

**Comment by tautschnig** [comment]
Commit series cleanup done. As reviewing at a diffstat of `717 files changed, 17618 insertions(+), 12827 deletions(-)` is rather impractical, please do take a look at the commit summaries and let me know which ones raise the most interest. I'll then factor these out into separate pull requests, unless you want to cherry-pick them (which may or may not work well).  - probably @peterschrammel, @kroening

**Comment by tautschnig** [comment]
So finally all checks are passing on this one.

**Comment by peterschrammel** [comment]
@tautschnig, this is on my list, but i will take a bit.

**Comment by tautschnig** [comment]
@peterschrammel Thanks a lot, much appreciated. Please, as said, do take the liberty to just browse the list of commit summaries and point at ones that you'd like to pick up first. I'll then happily prepare independent (and thus much easier to review) pull requests.

**Comment by tautschnig** [comment]
Seeing that each and every pull request includes linter-happiness fixes: any chance to receive feedback on this one? (Yes, all those linter-happiness fixes that people are putting in are already in this pull request...)

File: src/solvers/flattening/boolbv_mod.cpp [hunk 3]
```diff
@@ -24,6 +24,7 @@ bvt boolbvt::convert_mod(const mod_exprt &expr)
 {
   if(expr.type().id()==ID_floatbv)
   {
+    // TODO
```
Comment (peterschrammel) [inline src/solvers/flattening/boolbv_mod.cpp]: (just ignore, no assert or throw?)

File: src/solvers/flattening/boolbv_equality.cpp [hunk 4]
```diff
@@ -45,7 +45,7 @@ literalt boolbvt::convert_equality(const equal_exprt &expr)
     if(has_byte_operator(expr))
     {
       exprt tmp=flatten_byte_operators(expr, ns);
-      //std::cout << "X: " << from_expr(ns, "", tmp) << std::endl;
+      // std::cout << "X: " << from_expr(ns, "", tmp) << std::endl;
```
Comment (peterschrammel) [inline src/solvers/flattening/boolbv_equality.cpp]: #if 0 or #ifdef DEBUG

File: src/pointer-analysis/value_set_domain.cpp [hunk 5]
```diff
@@ -57,7 +57,9 @@ void value_set_domaint::transform(
     }
     break;
 
-  default:;
+  default:
+    {
+    }
     // do nothing
```
Comment (peterschrammel) [inline src/pointer-analysis/value_set_domain.cpp]: maybe move inside the braces

File: src/musketeer/propagate_const_function_pointers.cpp [hunk 6]
```diff
@@ -72,17 +85,20 @@ class const_function_pointer_propagationt {
     return true;
   }
 
-  bool remove(const irep_idt& symb) {
-    //assert(pointer_to_fun.find(symb)!=pointer_to_fun.end());
+  bool remove(const irep_idt& symb)
+  {
+    // assert(pointer_to_fun.find(symb)!=pointer_to_fun.end());
```
Comment (peterschrammel) [inline src/musketeer/propagate_const_function_pointers.cpp]: commented code (another issue we should tackle more generally)

File: src/goto-programs/show_properties.cpp [hunk 7]
```diff
@@ -45,7 +45,7 @@ void show_properties(
     const source_locationt &source_location=ins.source_location;
 
     const irep_idt &comment=source_location.get_comment();
-    //const irep_idt &function=location.get_function();
+    // const irep_idt &function=location.get_function();
```
Comment (peterschrammel) [inline src/goto-programs/show_properties.cpp]: (code commented out)

File: src/goto-instrument/accelerate/disjunctive_polynomial_acceleration.cpp [hunk 8]
```diff
@@ -200,7 +230,8 @@ bool disjunctive_polynomial_accelerationt::accelerate(
 
 
   /*
-  if (!utils.check_inductive(polynomials, assigns)) {
+  if (!utils.check_inductive(polynomials, assigns))
```
Comment (peterschrammel) [inline src/goto-instrument/accelerate/disjunctive_polynomial_acceleration.cpp]: #if 0

File: src/goto-instrument/accelerate/accelerate.cpp [hunk 9]
```diff
@@ -18,66 +26,76 @@
 #include "disjunctive_polynomial_acceleration.h"
 #include "overflow_instrumenter.h"
 #include "util.h"
-//#include "symbolic_accelerator.h"
+// #include "symbolic_accelerator.h"
```
Comment (peterschrammel) [inline src/goto-instrument/accelerate/accelerate.cpp]: #if 0 (many more)

File: src/analyses/invariant_set_domain.cpp [hunk 10]
```diff
@@ -80,7 +81,9 @@ void invariant_set_domaint::transform(
     invariant_set.make_threaded();
     break;
 
-  default:;
-    // do nothing
+  default:
+    {
+      // do nothing
```
Comment (tautschnig) [inline src/analyses/invariant_set_domain.cpp]: I'm not quite sure what you are suggesting here?

File: src/analyses/reaching_definitions.cpp [hunk 11]
```diff
@@ -827,8 +829,8 @@ bool rd_range_domaint::merge_shared(
   {
     const irep_idt &identifier=value.first;
 
-    if(!ns.lookup(identifier).is_shared() /*&&
-       !rd.get_is_dirty()(identifier)*/)
+    if(!ns.lookup(identifier).is_shared()
+       /*&& !rd.get_is_dirty()(identifier)*/)
```
Comment (tautschnig) [inline src/analyses/reaching_definitions.cpp]: I'm going to address this in a separate PR - the related TODO is a few lines above this commented bit.

File: src/analyses/natural_loops.h [hunk 12]
```diff
@@ -29,23 +29,23 @@ class natural_loops_templatet
...
-  inline void operator()(P &program)
+  void operator()(P &program)
   {
     compute(program);
   }
 
   void output(std::ostream &) const;
 
-  inline const cfg_dominators_templatet<P, T, false>& get_dominator_info() const
+  const cfg_dominators_templatet<P, T, false>& get_dominator_info() const
```
Comment (tautschnig) [inline src/analyses/natural_loops.h]: I'm going to add one commit to this PR that addresses all of those (736 across the code base).

File: src/analyses/invariant_set_domain.cpp (hunk 10 above)
Comment (peterschrammel) [inline src/analyses/invariant_set_domain.cpp]: Adding a comment inside the empty code block.

**Comment by tautschnig** [comment]
I believe I've now addressed all comments and have also fixed any issues that have crept into the codebase in the meantime.

**Comment by rjmunro** [comment]
@kroening Can this be merged?

**Pull request merged**